qa/tasks/mgr/dashboard/test_osd.py

```diff
@@ -260,10 +267,114 @@ def test_add_osd_flag(self):
             'sortbitwise', 'recovery_deletes', 'purged_snapdirs', 'noout',
             'pause', 'pglog_hardlimit'
         ])
-        self.assertEqual(flags, sorted([
+        self.assertEqual(sorted(flags), sorted([
             'sortbitwise', 'recovery_deletes', 'purged_snapdirs', 'noout',
             'pause', 'pglog_hardlimit'
         ]))
 
         # Restore flags
         self._put_flags(self._initial_flags)
+
+    def test_get_indiv_flag(self):
+        initial = self._get('/api/osd/flags/individual')
+        self.assertStatus(200)
+        self.assertSchema(initial, JList(JObj({
+            'osd': int,
+            'flags': JList(str)
+        })))
+
+        self._ceph_cmd(['osd', 'set-group', 'noout,noin', 'osd.0', 'osd.1', 'osd.2'])
+        flags_added = self._get('/api/osd/flags/individual')
+        self.assertStatus(200)
+        for osd in flags_added:
+            if osd['osd'] in [0, 1, 2]:
+                self.assertIn('noout', osd['flags'])
+                self.assertIn('noin', osd['flags'])
+                for osd_initial in initial:
+                    if osd['osd'] == osd_initial['osd']:
+                        self.assertGreater(len(osd['flags']), len(osd_initial['flags']))
+
+        self._ceph_cmd(['osd', 'unset-group', 'noout,noin', 'osd.0', 'osd.1', 'osd.2'])
+
+        flags_removed = self._get('/api/osd/flags/individual')
+        self.assertStatus(200)
+        for osd in flags_removed:
+            if osd['osd'] in [0, 1, 2]:
+                self.assertNotIn('noout', osd['flags'])
+                self.assertNotIn('noin', osd['flags'])
```

Comment:
These might be 2 different [subtests](https://docs.python.org/3.6/library/unittest.html#distinguishing-test-iterations-using-subtests)?

Reply:
Ok, I don't think this applies here. They're just 2 steps in a longer tests, right?

Reply:
Yes. It's about:

1. Get the initial situation
2. Set some individual flags and check if the GET response is correct
3. Remove the individual flags (to reset to the initial situation) and check if the GET response is correct as well